src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionResolver.cs

```diff
@@ -158,7 +158,7 @@ public override async Task<WorkspaceEdit> ResolveAsync(JObject data, Cancellatio
             string newUsingNamespace)
         {
             // If we don't have usings, insert after the last namespace or page directive, which ever comes later
-            var head = new Position(1, 0);
+            var head = new Position(0, 0);
```

Comment:
This might be causing side effects. We do want `@using` directives to appear after a `@page` directive if it exists like the comment on line 160 says. I suspect the correct fix might be a little more complicated than this.

Reply:
![FixAddUsingLinePage](https://user-images.githubusercontent.com/14852843/90685785-291f4480-e21f-11ea-8012-4b9a1f99fc53.gif)

Not sure I follow. The `@using` is added after the `@page`.

Reply:
How is that position 0,0? I'm confused. Maybe I am missing something.

Reply:
https://github.com/dotnet/aspnetcore-tooling/pull/2414/files#diff-3562bd0503d5e8caa2d99533c55d5e02R169 should be handling the additional offset.

Reply:
https://github.com/dotnet/aspnetcore-tooling/blob/e8fea86d36b0cfc17439458d8d0a02de2d182afc/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionResolver.cs#L163-L169

First link doesn't seem to go to the right location.

Reply:
Not sure what line that link is pointing to. It doesn't navigate properly for me.

Reply:
Ahh this is just the initial value that was wrong. Got it 👍 

I highly recommend moving that comment down to where we are actually computing that offset.

Reply:
Done! https://github.com/dotnet/aspnetcore-tooling/pull/2414/commits/7baaf0a49705f8c14e3a6dac59f6ab0a77bcbb03